### PR TITLE
Added the missing data to the basque fiscalization signature

### PR DESCRIPTION
**Pull request by abdallahbeshi**
## Description

The signature must include more data like (Signing time, algo, sig policy per region, sig policy link, issuer name...etc).

Fixes #issue

## Type of change
- [ ] Bug fix.
- [ ] Feature.
- [x] Consolidation.

## Checklist
- [ ] Is breaking change.
- [ ] Documentation updated.
- [ ] Tests included (please specify cases).

### CODE REVIEW

File: src/Basque/Mews.Fiscalizations.Basque/TicketBaiSignature.cs [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            keyInfo.AddClause(certificateData);
+            signedXml.KeyInfo = keyInfo;
+        }
+
+        private static void AddReferences(XAdESSignedXml signedXml)
+        {
+            var reference1 = new Reference
+            {
+                Type = "",
+                Uri = $"",
```
Comment (KaliCZ) [inline src/Basque/Mews.Fiscalizations.Basque/TicketBaiSignature.cs:129]: So what is this? 😄

**Review by KaliCZ — COMMENTED** [review]
Generally speaking looks alright, but to validate this, somebody needs to read the documentation. So pls talk to @marioporcic or @tonyjoanes. Or even both.

File: src/Basque/Mews.Fiscalizations.Basque/TicketBaiClient.cs [hunk 2]
```diff
@@ -39,46 +39,25 @@ public async Task<SendInvoiceResponse> SendInvoiceAsync(SendInvoiceRequest reque
                 encoding: ServiceInfo.Encoding,
                 namespaces: NonEmptyEnumerable.Create(new XmlNamespace("http://www.w3.org/2000/09/xmldsig#"))
             ));
-            SignXml(xmlDoc.OwnerDocument);
+            TicketBaiSignature.Sign(xmlDoc.OwnerDocument, Certificate, Region);
```
Comment (tonyjoanes) [inline src/Basque/Mews.Fiscalizations.Basque/TicketBaiClient.cs:42]: Are we happy that the tests cover the new stuff that is now happening in this method?

**Review by tonyjoanes — CHANGES_REQUESTED** [review]

File: src/Basque/Mews.Fiscalizations.Basque/TicketBaiSignature.cs [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            sigPolicyDescription.InnerText = "TicketBAI sinadura-politika / Politica de firma TicketBAI";
+
+            var sigPolicyHash = xmlDoc.CreateElement(XadesPrefix, "SigPolicyHash", NamespaceUri);
+
+            var sigPolicyHashDigestMethod = xmlDoc.CreateElement(DsigPrefix, "DigestMethod", NamespaceUri);
+            sigPolicyHashDigestMethod.SetAttribute("Algorithm", SignedXml.XmlDsigSHA256Url);
+
+            var sigPolicyHashDigestValue = xmlDoc.CreateElement(DsigPrefix, "DigestValue", NamespaceUri);
+            sigPolicyHashDigestValue.InnerText = region.Match(
+                Region.Gipuzkoa, _ => "vSe1CH7eAFVkGN0X2Y7Nl9XGUoBnziDA5BGUSsyt8mg=",
```
[the remark this answers is not in the document]
Comment (abdallahbeshi) [reply src/Basque/Mews.Fiscalizations.Basque/TicketBaiSignature.cs:59]: Nope, they shouldn't ever change

File: src/Basque/Mews.Fiscalizations.Basque/TicketBaiSignature.cs [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+            signedXml.AddObject(dataObject);
+            signedXml.SigningKey = certificate.PrivateKey;
+
+            AddKeyInfo(signedXml, certificate);
+            AddReferences(signedXml);
+
+            signedXml.ComputeSignature();
+
+            var xmlDigitalSignature = signedXml.GetXml();
+            xmlDoc.DocumentElement.AppendChild(xmlDoc.ImportNode(xmlDigitalSignature, deep: true));
```
[the remark this answers is not in the document]
Comment (abdallahbeshi) [reply src/Basque/Mews.Fiscalizations.Basque/TicketBaiSignature.cs:110]: nope, its not nullable

File: src/Basque/Mews.Fiscalizations.Basque/TicketBaiClient.cs (hunk 2 above)
Comment (abdallahbeshi) [reply src/Basque/Mews.Fiscalizations.Basque/TicketBaiClient.cs:42]: Sure, we can have a ticket for writing some tests to verify the data in the XML.

Comment (KaliCZ) [reply src/Basque/Mews.Fiscalizations.Basque/TicketBaiClient.cs:42]: Why a ticket? A ticket just means we don't have to do it now. But why not do it now? Why should we code something and deploy it without having the tests first?

**Pull request closed without merging**